plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

```diff
@@ -232,30 +237,98 @@ const { actions, state } = store< VariableProductAddToCartWithOptionsStore >(
 				}
 			},
 			handlePillClick() {
-				if ( state.isOptionDisabled ) {
-					return;
-				}
 				const context = getContext< Context >();
+
 				if ( context.selectedValue === context.option.value ) {
 					context.selectedValue = '';
 				} else {
 					context.selectedValue = context.option.value;
 				}
 				actions.setAttribute( context.name, context.selectedValue );
+				actions.autoselectOtherAttributes();
 			},
 			handleDropdownChange( event: ChangeEvent< HTMLSelectElement > ) {
 				const context = getContext< Context >();
 				context.selectedValue = event.currentTarget.value;
 				actions.setAttribute( context.name, context.selectedValue );
+				actions.autoselectOtherAttributes();
+			},
+			autoselectAttributes( variation_selectors: Element[] ) {
+				const { selectedAttributes } = getContext< Context >();
+
+				for ( const current_variation_selector of variation_selectors ) {
+					const optionStyle = current_variation_selector.dataset?.optionStyle || 'pills';
+					// Dropdown options or Pill inputs,
+					// that HAVE a value (Choose an Option has an empty value of ""),
+					// and are compatible with the possible variations
+					let valid_choices: Element[] = [];
+					current_variation_selector.querySelectorAll( 'option:not([value=""]), input.wc-block-add-to-cart-with-options-variation-selector-attribute-options__pill-input:not([value=""])' ).forEach( ( el ) => {
```

Comment:
There are a few variables in this file that will need to be converted into camel case.